### PR TITLE
 Bring determinism to your programs

### PR DESCRIPTION
**Pull request by kumare3**
- Durable Sleep: Sleep once and maintain consistency in the sleep
- Durable time, simple wrapper to get the same time again.

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

File: tests/flyte/extras/test_frozen_time.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    with (
+        patch("flyte.durable._time.time.time", side_effect=[100.0, 103.0]),
+        patch("flyte.durable._time.asyncio.sleep", new_callable=AsyncMock) as mock_sleep,
+    ):
+        await durable_sleep.aio(5.0)
+        mock_sleep.assert_awaited_once_with(2.0)
+
+
+@pytest.mark.asyncio
+async def test_durable_sleep_zero_seconds():
```
Comment (machichima) [inline tests/flyte/extras/test_frozen_time.py:106]: Should we also add test to cover negative value for `durable_sleep`?

File: tests/flyte/extras/test_frozen_time.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    assert isinstance(result, float)
+
+
+def test_durable_time_sync_returns_float():
+    result = durable_time()
+    assert isinstance(result, float)
+
+
+@pytest.mark.asyncio
+async def test_durable_time_returns_current_time():
```
Comment (machichima) [inline tests/flyte/extras/test_frozen_time.py:26]: I think for `durable_time`, we want it to give consistent value for all replays? Is it possible for us to add a test for replay scenario?

Same comment for `durable_now`

File: tests/flyte/extras/test_frozen_time.py (hunk 1 above)
Comment (kumare3) [reply tests/flyte/extras/test_frozen_time.py:106]: thats ok

File: tests/flyte/extras/test_frozen_time.py (hunk 2 above)
Comment (kumare3) [reply tests/flyte/extras/test_frozen_time.py:26]: hmm let me think, i will merge and then comeback
